addons/point_of_sale/models/pos_bus_mixin.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import uuid
+from odoo import fields, models
+
+class PosBusMixin(models.AbstractModel):
+    _name = "pos.bus.mixin"
+    _description = "Bus Mixin"
+
+    access_token = fields.Char('Security Token', readonly=True, default=lambda self: str(uuid.uuid4()))
```

Comment:
Yes, you are right @xmo-odoo , this makes sense. The portal mixin does that with the `_portal_ensure_token` method and at first i wanted to do the same thing, but i wanted to prevent the case where someone tries to access the access_token field which has not already been computed. Is there a solution to this problem? Something similar to how we would write a getter in js?